externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala

```diff
@@ -147,6 +162,34 @@ case class SparkSQLEngine(spark: SparkSession) extends Serverable("SparkSQLEngin
         TimeUnit.MILLISECONDS)
     }
   }
+
+  private[kyuubi] def startFailFastChecker(stop: () => Unit): Unit = {
+    val interval = conf.get(ENGINE_CHECK_INTERVAL)
+    val deregistered = new AtomicBoolean(false)
+    var timeout = Duration(60, TimeUnit.SECONDS).toMillis
```

Comment:
Using a fixed duration 60 seconds, but I am open to add a new configuration key.

Reply:
60s may be fine, but I think it should be configurable.

Reply:
Got it, I will add it in the next commit.